src/ProxyClient/Cloudflare.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+use Http\Message\RequestFactory;
+
+/**
+ * Cloudflare HTTP cache invalidator.
+ *
+ * Additional constructor options:
+ * - zone_identifier       Identifier for your Cloudflare zone you want to purge the cache for
+ * - authentication_token  API authorization token, requires Zone.Cache Purge permissions
+ *
+ * @author Simon Jones <simon@studio24.net>
+ */
+class Cloudflare extends HttpProxyClient implements ClearCapable, PurgeCapable, TagCapable
+{
+    /**
+     * @see https://api.cloudflare.com/#getting-started-endpoints
+     */
+    private const API_ENDPOINT = '/client/v4';
+
+    /**
+     * Batch URL purge limit.
+     *
+     * @see https://api.cloudflare.com/#zone-purge-files-by-url
+     */
+    private const URL_BATCH_PURGE_LIMIT = 30;
+
+    /**
+     * Array of data to send to Cloudflare for purge by URLs request.
+     *
+     * To reduce the number of requests to cloudflare, we buffer the URLs.
+     * During flush, we build requests with batches of URL_BATCH_PURGE_LIMIT.
+     *
+     * @var array<string|array{url: string, headers: string[]}>
+     */
+    private $purgeByUrlsData = [];
+
+    public function __construct(
+        Dispatcher $httpDispatcher,
+        array $options = [],
+        RequestFactory $messageFactory = null
+    ) {
+        if (!function_exists('json_encode')) {
```

Comment:
i added a constructor to check about ext-json, because it is a "soft-dependency" and not explicitly noted in the composer.json (and most clients don't need json, but this one and fastly do)